llvm/lib/Target/X86/X86AsmPrinter.cpp

```diff
@@ -55,6 +55,116 @@ X86AsmPrinter::X86AsmPrinter(TargetMachine &TM,
 // Primitive Helper Functions.
 //===----------------------------------------------------------------------===//
 
+const TargetInstrInfo *TII;
+
+/// Clear any mappings that map to the given register.
+void clearRhs(Register Reg, std::map<Register, int64_t> &SpillMap) {
+  for (auto I = SpillMap.begin(); I != SpillMap.end();) {
```

Comment:
I tend to prefer this style of `for` as a `while` because it makes it clearer that not every iteration is incrementing `I`.

Reply:
I only had to write it this way because we are erasing things from the map. So technically that erasing case still "iterates" `I`. But I'm happy to change it if you feel strongly about it.

Reply:
I feel moderately strongly about it, mostly because I've burnt myself with odd `for` loops in the past!

Reply:
Here ya go: [8c128bb](https://github.com/ykjit/ykllvm/pull/57/commits/8c128bb069131b7a579a9b693f170a3b46fed572)